neqo-transport/src/cc/classic_cc.rs

```diff
@@ -978,131 +986,184 @@ mod tests {
 
     #[test]
     fn app_limited_slow_start() {
-        const LESS_THAN_CWND_PKTS: usize = 4;
+        const BELOW_APP_LIMIT_PKTS: usize = 5;
+        const ABOVE_APP_LIMIT_PKTS: usize = BELOW_APP_LIMIT_PKTS + 1;
         let mut cc = ClassicCongestionControl::new(NewReno::default());
-
-        for i in 0..CWND_INITIAL_PKTS {
-            let sent = SentPacket::new(
-                PacketType::Short,
-                u64::try_from(i).unwrap(), // pn
-                now(),                     // time sent
-                true,                      // ack eliciting
-                Vec::new(),                // tokens
-                MAX_DATAGRAM_SIZE,         // size
-            );
-            cc.on_packet_sent(&sent);
+        let cwnd = cc.congestion_window;
+        let mut now = now();
+        let mut next_pn = 0;
+
+        // simulate packet bursts below app_limit
+        for packet_burst_size in 1..=BELOW_APP_LIMIT_PKTS {
+            // always stay below app_limit during sent.
+            let mut pkts = Vec::new();
+            for _ in 0..packet_burst_size {
+                let p = SentPacket::new(
+                    PacketType::Short,
+                    next_pn,           // pn
+                    now,               // time sent
+                    true,              // ack eliciting
+                    Vec::new(),        // tokens
+                    MAX_DATAGRAM_SIZE, // size
+                );
+                next_pn += 1;
+                cc.on_packet_sent(&p);
+                pkts.push(p);
+            }
+            assert_eq!(cc.bytes_in_flight(), packet_burst_size * MAX_DATAGRAM_SIZE);
+            now += RTT;
+            for (i, pkt) in pkts.into_iter().enumerate() {
+                cc.on_packets_acked(&[pkt], RTT, now);
```

Comment:
Is there any reason you couldn't do this instead?

```suggestion
            cc.on_packets_acked(&pkts, RTT, now);
```

The assertions would get easier as well; you would just check that bytes_in_flight is 0 and the cwnd hasn't grown.

I suggest this because it's a piece of the test that you aren't really interested in; we're still in a setup phase.